stl/inc/vector

```diff
@@ -969,22 +1165,36 @@ public:
         pointer& _Mylast  = _My_data._Mylast;
 
         _My_data._Orphan_all();
+        const auto _Oldcapacity = static_cast<size_type>(_My_data._Myend - _Myfirst);
+        if (_Newsize > _Oldcapacity) { // reallocate
+            _Clear_and_reserve_geometric(_Newsize);
+            if constexpr (is_nothrow_copy_constructible_v<_Ty>) {
+                _Mylast = _Uninitialized_fill_n(_My_data._Myfirst, _Newsize, _Val, _Al);

```

Comment:
Note: there were some final remarks from @CaseyCarter on the fix vector PR, which need to be incorporated here

Reply:
See https://github.com/microsoft/STL/pull/2161/commits/5f7fb298c8bae298a150680929eadf64e54d45a5